src/napari_spatialdata/utils/_utils.py

```diff
@@ -367,3 +367,18 @@ def get_elements_meta_mapping(
                 name_to_add = name
             elements[name] = elements_metadata
     return elements, name_to_add
+
+
+def _get_metadata_adata(sdata: SpatialData, key: str) -> None | AnnData:
```

Comment:
Nice to have this separation, it makes the code more readable. Since there is a bug with the order of the rows (see my comment in the discussion), we could fix it by replacing this function with `match_table_to_element()` https://github.com/scverse/spatialdata/blob/bf3377fd36f94ef51a233ae71312b1c28e978e6e/src/spatialdata/_core/query/relational_query.py#L134, which returns a table with the correct order.